src/cbmc_starter_kit/template-for-repository/proofs/lib/summarize.py

```diff
@@ -0,0 +1,113 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import json
+import logging
+import sys
+
+
+def _get_max_length(column):
+    return max(len(word) for word in column)
+
+
+def _get_list_of_max_length_per_column(data):
+    list_of_max_length_per_column = []
```

Comment:
Not going to block on this but I found this a bit difficult to read. What do you think, is this cleaner? (also removes the need for the function above)
```
ret = [len(item) for item in data[0]]
for row in data[1:]:
  for idx, item in enumerate(row):
      ret[idx] = max(ret[idx], len(item))
return ret
```

Reply:
If you want to go overboard you could even replace the inner loop with a comprehension which would save another line, depending on whether you think that would be easier to read.

Reply:
To be honest I prefer this style, but I realise nested comprehensions are sometimes difficult to read
```
ret = [len(item) for item in data[0]]
return [[max(ret[idx], len(item)) for idx, item in row] for row in data[1:]]
```

Reply:
I went with your first suggestion which indeed increased readability of the original function.